snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -513,14 +525,28 @@ def _generate_desktop_file(self, name, app):
             desktop_file.parse_and_reformat()
             desktop_file.write(gui_dir=os.path.join(self.meta_dir, 'gui'))
 
-    def _render_socket_modes(self, apps):
+    def _render_socket_modes(self, apps: Dict[str, Any]) -> None:
         for app in apps.values():
             sockets = app.get('sockets', {})
             for socket in sockets.values():
                 mode = socket.get('socket-mode')
                 if mode is not None:
                     socket['socket-mode'] = OctInt(mode)
-        return apps
+
+    def _apply_passthrough(self, location: str, section: Dict[str, Any],
+                           passthrough: Dict[str, Any],
+                           original: Dict[str, Any]) -> None:
+        # Any value already in the original dictionary must
+        # not be specified in passthrough at the same time.
+        duplicates = set(original.keys() & passthrough.keys())
+        if duplicates:
+            raise meta_errors.AmbiguousPassthroughKeyError(duplicates)
+        section.update(passthrough)
+        if passthrough:
+            logger.warn('Passthrough is being used to add experimental '
```

Comment:
Our thread is considered "outdated" by github, so I'm gonna copy/paste and start a new one :stuck_out_tongue: .

I think the real goal is to let them know that, because this keyword is being used, the store will flag it. Why do we need to say that multiple times? If you want to say where it was used, that's fine, but let's not print this more than once either way.